rpmio/rpmkeyring.c

```diff
@@ -288,7 +346,15 @@ rpmRC rpmKeyringVerifySig(rpmKeyring keyring, pgpDigParams sig, DIGEST_CTX ctx)
 	    rpmlog(rc ? RPMLOG_ERR : RPMLOG_WARNING, "%s\n", lints);
 	    free(lints);
 	}
+	if (keyptr) {
+	    *keyptr = rpmPubkeyLink(key);
```

Comment:
Yesterday's talk about the backwards way subkeys are added made me realize another thing:
Since all we care about is the fingerprint of the primary key, why are we sending it subkeys to begin with?
This all should be a lot, lot easier if we just toss in the primary key from here, don't even need the public rpmPubkeyPrimaryKey() API then.

Reply:
Yup. I was aligning this with other sub key handling which was very public. What about just keeping this PR as is and cleaning this up as part of #3350?

Reply:
OK, just made `pubkeyPrimarykey` private and only return primary keys as suggested.